find-trio-denovo.py

```diff
@@ -14,167 +13,105 @@
-				#It still works, but might be more than what's necessary
-
-				# Variant conditions if the data is phased
-				if delim == "|":
-					if child1_in_dad == 0 and child2_in_mom == 0:
-						recordVariant(line) # variant because neither child allele in parents
-						variant_count +=1
-					elif child1_in_dad == 0 or child2_in_mom == 0:	
-						recordVariant(line) #variant because one parent didn't contribute an allele
-						variant_count +=1
-					
-				# Variant conditions if the data is unphased
-				# each child allele can now come from either parent, so it's a variant only if it's not in either parent
-				# before, only checked first child allele against dad. now need to check against mom also.
-					
-				elif delim == "/":   #can just be "else"  Wrote it out to make it clear for now
-					if child1_in_dad == 0 and child1_in_mom == 0:
-						recordVariant(line) #variant because child1 not in either parent
-						unphasedVariants(line) #also write to second output file, where all lines are treated as unphased
-						variant_count +=1
-						unphased_count +=1
-					elif child2_in_mom == 0 and child2_in_dad == 0:
-						recordVariant(line) #variant because child2 not in either parent
-						unphasedVariants(line)
-						variant_count +=1
-						unphased_count +=1
-					elif child1_in_dad == 0 and child2_in_dad ==0:
-						recordVariant(line) #variant because neither child allele in dad
-						unphasedVariants(line)
-						variant_count +=1
-						unphased_count +=1
-					elif child1_in_mom == 0 and child2_in_mom ==0:
-						recordVariant(line) #variant because neither child allele in mom
-						unphasedVariants(line)
-						variant_count +=1
-						unphased_count +=1
-				
-				#Treat phased lines as unphased and record variants in second output file
-				if delim == "|":   
-					if child1_in_dad == 0 and child1_in_mom == 0:
-						unphasedVariants(line) 
-						unphased_count +=1
-					elif child2_in_mom == 0 and child2_in_dad == 0:
-						unphasedVariants(line)
-						unphased_count +=1
-					elif child1_in_dad == 0 and child2_in_dad ==0:
-						unphasedVariants(line)
-						unphased_count +=1
-					elif child1_in_mom == 0 and child2_in_mom ==0:
-						unphasedVariants(line)
-						unphased_count +=1
-
-	recordVariant("\n")			
-	recordVariant('The script took {0} minutes'.format( (time.time() - startTime)/60 ) )
-	recordVariant("\n")
-	recordVariant(str(variant_count) + " phased and unphased variants were found")
-
-	unphasedVariants("\n")	
-	unphasedVariants('The script took {0} minutes'.format( (time.time() - startTime)/60 ) )
-	unphasedVariants("\n")
-	unphasedVariants(str(unphased_count) + " variants were found when all lines were treated as unphased")
+	with open (inFileName, 'r') as infile:  #when you use "with open" you don't have to close the file later
+            with open (inFileName + ".variants", "w") as variantFile:
+                with open (inFileName + ".variants.as.unphased", "w") as unphasedFile:
```

Comment:
Your comment about using `open` shows it's advantages. Below you had functions that open and close a file and append a line to it. This works, but it's annoying to have to clear out the file between runs to see the true results. Using `open with` at the top level here allows us to truncate the file automatically (via the `'w'` flag).